src/utils/bigint.ts

```diff
@@ -0,0 +1,169 @@
+
+      if (source.includes('.')) {
+        // If value is an integer and contains a '.', it must be like '123.0', so we extract the integer part only.
+        let zeroes: string;
+        [source, zeroes] = source.split('.');
+
+        if (zeroes.split('').some(char => char !== '0')) {
+          // This case shouldn't happen but we'll prohibit it to be safe. For example, if the source is
+          // '12345678901234567890.1', JS will parse it as an integer with loss of precision, `12345678901234567000`.
+          throw Error(`large float will lose precision! in "${text}"`);
+        }
+      }
+
+      const bigIntValue = BigInt(source);
+      if (bigIntValue !== BigInt(value)) {
+        // If the parsed value is an integer and its BigInt representation is a different value,
+        // it means we lost precision, so we return the BigInt.
+        return bigIntValue;
+      }
+
+      // No special handling needed.
+      return value;
+    }
+
+    // @ts-expect-error TypeScript hasn't been updated with the `context` argument from Node v22.
+    return JSON.parse(text, this.isAvailable() ? bigIntReviver : undefined);
+  },
+
+  stringify(value: any, space?: string | number): string {
+    function bigIntReplacer(_key: string, value_: any): any {
+      // If the value is a BigInt, we simply return its string representation.
+      // @ts-expect-error TypeScript hasn't been updated with the `rawJSON` function from Node v22.
+      return typeof value_ === 'bigint' ? JSON.rawJSON(value_.toString()) : value_;
+    }
+
+    return JSON.stringify(value, this.isAvailable() ? bigIntReplacer : undefined, space);
+  },
+  /* eslint-enable @typescript-eslint/no-explicit-any */
+
+  /**
+   * Utility function for checking if JSONBigInt is available. We shouldn't allow it to be used if it's not available,
+   * however we temporarily keep Node v20 in our CI, so we have to allow it for tests.
+   * After QA is done to test Node v22 and we remove v20 from CI, we may remove this function.
+   */
+  isAvailable(): boolean {
+    const major = process.versions.node.split('.')[0];
+    return Number(major) >= 22;
+  },
+};
+
+/**
+ * A utility Zod schema for `bigint` properties that can be instantiated from a coercible type, that is, a `number`, `string`, or `bigint` itself.
+ */
+export const bigIntCoercibleSchema = z
+  .bigint()
+  .or(z.number())
+  .or(z.string())
+  .pipe(z.coerce.bigint());
+
+/**
+ * A type alias for a Zod schema with `unknown` input and generic output.
+ */
+export type ZodSchema<T> = z.ZodSchema<T, z.ZodTypeDef, unknown>;
+
+/**
+ * Parse some `unknown` data with a Zod schema. If parsing fails, it logs the error and throws.
+ */
+export function parseSchema<T>(data: unknown, schema: ZodSchema<T>): T {
+  const result = schema.safeParse(data);
+
+  if (!result.success) {
+    // TODO: How to log correctly?
+    // eslint-disable-next-line no-console
+    console.error(`error: ${result.error.message}\ncaused by input: ${data}`);
```

Comment:
What's the correct logger to use here? `getDefaultLogger()`?

Reply:
> Is this an unexpected error or it can happen sometimes?

It can happen sometimes, this function receives an untyped object and tries to parse it using the provided schema. If the object does not conform, the error will be thrown. This can happen, for example, if the full node returns an incorrect response schema in the API.

> And yes, I think the `getDefaultLogger()` is the best one to use

Changed in https://github.com/HathorNetwork/hathor-wallet-lib/pull/728/commits/9ef97a7156a03be8b6da0a68f85009d6ec615d11